### PR TITLE
Makes consist add order-preserving

### PR DESCRIPTION
**Pull request by balazsracz**
Updates the linked lists for consists to be order-preserving.
Updates consist add return value to not fail when a consist flag is being updated.

### CODE REVIEW

File: src/openlcb/TractionTrain.cxx [hunk 1]
```diff
@@ -369,10 +369,14 @@ struct TrainService::Impl
                             Defs::ERROR_INVALID_ARGS_MESSAGE_TOO_SHORT);
                     NodeID target = data_to_node_id(payload() + 3);
                     uint8_t flags = payload()[2];
-                    bool resp = train_node()->add_consist(target, flags);
+                    if (target == train_node()->node_id()) {
```
Comment (bakerstu) [inline src/openlcb/TractionTrain.cxx]: style:  hanging brace.

File: src/openlcb/TractionTrain.hxx [hunk 2]
```diff
@@ -130,15 +130,16 @@ public:
     {
         if (!tgt) return false;
         if (tgt == node_id()) return false;
```
Comment (bakerstu) [inline src/openlcb/TractionTrain.hxx]: style:  both if statements above require braces

File: src/openlcb/TractionTrain.cxx (hunk 1 above)
Comment (balazsracz) [reply src/openlcb/TractionTrain.cxx]: fixed

File: src/openlcb/TractionTrain.hxx (hunk 2 above)
Comment (balazsracz) [reply src/openlcb/TractionTrain.hxx]: done

**Pull request merged**